src/analysis/processing/qgsalgorithmtransect.cpp

```diff
@@ -0,0 +1,182 @@
+/***************************************************************************
+                         qgsalgorithmtransect.cpp
+                         -------------------------
+    begin                : October 2017
+    copyright            : (C) 2017 by Loïc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmtransect.h"
+
+///@cond PRIVATE
+
+QString QgsTransectAlgorithm::name() const
+{
+  return QStringLiteral( "transect" );
+}
+
+QString QgsTransectAlgorithm::displayName() const
+{
+  return QObject::tr( "Transect" );
+}
+
+QStringList QgsTransectAlgorithm::tags() const
+{
+  return QObject::tr( "transect,line,extend," ).split( ',' );
```

Comment:
How about adding "station" and changing "line" to "lines", so that users of your plugin will also find this alg easily